components/Modal.jsx

```diff
@@ -0,0 +1,194 @@
+  if (error) return <p>Error loading keycaps...</p>;
+  if (!keycaps) return <p> Loading keycaps...</p>;
+
+  const selectedKeycapObj = keycaps.find(
+    (keycap) => keycap.name === selectedKeycap
+  );
+
+  const kitsAvailable =
+    selectedKeycapObj?.kits?.flatMap((kit) => kit.price_list) ?? [];
+
+  // Code for Checkbox Selection
+  const handleKitSelection = (event) => {
+    const { value, checked } = event.target;
+    setSelectedKits(
+      (prevSelected) =>
+        checked
+          ? [...prevSelected, value] // Add kit
+          : prevSelected.filter((kit) => kit !== value) // Remove kit
+    );
+  };
+
+  return createPortal(
+    <>
+      <OverlayDiv />
+      <ModalHolder>
+        <h2>Select a Keycap Set</h2>
+
+        {/* Dropdown: Select Keycap Set */}
+        <DropDownSelect
+          value={selectedKeycap}
+          onChange={(e) => {
+            setSelectedKeycap(e.target.value);
+            setSelectedKits([]); // Reset kit selection
+          }}
+        >
+          <option value="">-- Choose a keycap set --</option>
+
+          {keycaps.map((keycap) => (
+            <option key={keycap._id} value={keycap.name}>
+              {keycap.name}
+            </option>
+          ))}
+        </DropDownSelect>
+
+        {/* Checkbox Selection for Kits */}
+        {kitsAvailable?.length > 0 ? (
+          <>
+            <h3>Available Kits</h3>
+            <CheckboxContainer>
+              {kitsAvailable.map((kit) => (
+                <CheckboxLabel key={kit._id || kit.name}>
+                  <input
+                    type="checkbox"
+                    value={kit.name}
+                    checked={selectedKits.includes(kit.name)}
+                    onChange={handleKitSelection}
+                  />
+                  {kit.pic && (
+                    <Image
+                      src={kit.pic}
+                      alt={kit.name}
+                      width={50}
+                      height={50}
+                      style={{
+                        objectFit: "cover",
+                        marginRight: "10px",
+                        borderRadius: "5px",
+                      }}
+                    />
+                  )}
+                  {kit.name}
+                </CheckboxLabel>
+              ))}
+            </CheckboxContainer>
+          </>
+        ) : selectedKeycap ? (
+          <p>No kits available for this set.</p>
+        ) : null}
+
+        {/* Display Selected Items */}
+
+        {selectedKeycap && (
+          <p>
+            <strong>Set:</strong> {selectedKeycap}
+          </p>
+        )}
+        {selectedKits.length > 0 && (
+          <p>
+            <strong>Selected Kits:</strong> {selectedKits.join(", ")}
+          </p>
+        )}
+
+        <CancelButton onClick={onClose}>Cancel</CancelButton>
+        <AddButton
+          onClick={() =>
+            selectedKeycapObj && (onAddKeycap(selectedKeycapObj._id), onClose())
+          }
```

Comment:
please refactor into two lines of code
```suggestion
          onClick={() => {
            if(!selectedKeycapObj)return;
            
            onAddKeycap(selectedKeycapObj._id); 
            onClose();
            }
          }
```